src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverService.java

```diff
@@ -21,20 +23,37 @@
 
     private final ServiceJurisdictionMappingConfig serviceJurisdictionMappingConfig;
 
+    private final EnvelopeProcessor envelopeProcessor;
+
     public EnvelopeRetrieverService(
         EnvelopeRepository envelopeRepository,
-        ServiceJurisdictionMappingConfig serviceJurisdictionMappingConfig
+        ServiceJurisdictionMappingConfig serviceJurisdictionMappingConfig,
+        EnvelopeProcessor envelopeProcessor
     ) {
         this.envelopeRepository = envelopeRepository;
         this.serviceJurisdictionMappingConfig = serviceJurisdictionMappingConfig;
+        this.envelopeProcessor = envelopeProcessor;
     }
 
-    public List<Envelope> getAllEnvelopesForJurisdiction(String serviceName) {
+    public List<Envelope> getProcessedEnvelopesByJurisdiction(String serviceName) {
         String jurisdiction = getJurisdictionByServiceName(serviceName);
 
-        log.info("Fetching all envelopes for service {} and jurisdiction {}", serviceName, jurisdiction);
+        log.info("Fetching all processed envelopes for service {} and jurisdiction {}", serviceName, jurisdiction);
+
+        List<Envelope> processedEnvelopes =
+            envelopeRepository.findByJurisdictionAndStatus(
+                jurisdiction,
+                Event.DOC_PROCESSED
+            );
+
+        updateEnvelopeStatusAndCreateEvent(processedEnvelopes);
+
+        return processedEnvelopes;
+    }
 
-        return envelopeRepository.findByJurisdiction(jurisdiction);
+    private void updateEnvelopeStatusAndCreateEvent(List<Envelope> processedEnvelopes) {
+        processedEnvelopes.forEach(envelopeProcessor::markAsConsumed);
+        envelopeRepository.saveAll(processedEnvelopes);
```

Comment:
something does not look right here. `processedEnvelopes` are not updated. besides, all envelopes are updated in the `markAsConsumed` function